topostats/tracing/pruning.py

```diff
@@ -321,62 +334,29 @@ def _prune_by_length(  # pylint: disable=too-many-locals  # noqa: C901
         npt.NDArray
             Pruned skeleton as binary array.
         """
-        pruning = True
-        while pruning:
-            single_skeleton = rm_nibs(single_skeleton)
-            n_branches = 0
-            coordinates = np.argwhere(single_skeleton == 1).tolist()
-
-            # The branches are typically short so if a branch is longer than
-            #  0.15 * total points, its assumed to be part of the real data
-            max_branch_length = max_length if max_length != -1 else int(len(coordinates) * 0.15)
-            LOGGER.info(f"[pruning] : Maximum branch length : {max_branch_length}")
-            # first check to find all the end coordinates in the trace
-            potential_branch_ends = self._find_branch_ends(coordinates)
-
-            # Now check if its a branch - and if it is delete it
-            for branch_x, branch_y in potential_branch_ends:
-                branch_coordinates = [[branch_x, branch_y]]
-                branch_continues = True
-                temp_coordinates = coordinates[:]
-                temp_coordinates.pop(temp_coordinates.index([branch_x, branch_y]))
-
-                while branch_continues:
-                    n_neighbours, neighbours = genTracingFuncs.count_and_get_neighbours(
-                        branch_x, branch_y, temp_coordinates
-                    )
-
-                    # If branch continues
-                    if n_neighbours == 1:
-                        branch_x, branch_y = neighbours[0]
-                        branch_coordinates.append([branch_x, branch_y])
-                        temp_coordinates.pop(temp_coordinates.index([branch_x, branch_y]))
-
-                    # If the branch reaches the edge of the main trace
-                    elif n_neighbours > 1:
-                        branch_coordinates.pop(branch_coordinates.index([branch_x, branch_y]))
-                        branch_continues = False
-                        is_branch = True
-
-                    # Weird case that happens sometimes (would this be linear mols?)
-                    elif n_neighbours == 0:
-                        is_branch = True
-                        branch_continues = False
-
-                    # why not `and branch_continues`?
-                    if len(branch_coordinates) > max_branch_length:
-                        branch_continues = False
-                        is_branch = False
-                #
-                if is_branch:
-                    n_branches += 1
-                    for x, y in branch_coordinates:
-                        single_skeleton[x, y] = 0
-
-            if n_branches == 0:
-                pruning = False
-
-        return single_skeleton
+        coordinates = np.argwhere(single_skeleton == 1).tolist()
+        # The branches are typically short so if a branch is longer than
+        #  0.15 * total points, its assumed to be part of the real data
```

Comment:
We should try and make our functions/methods as flexible as possible and avoid hard-coding variables. The default can be set as the default argument to the option but it then makes it considerably easier in the future when someone decides they want to use a threshold of say 10% of total branch length.

This would require adding another argument such as `max_length_proportion: float = 0.15`.

Reply:
Honestly I may just fully remove the %-based functionality for favour of nm length-based pruning as it has led to some confusion among users and the iterative pruning can cause some possibly unexpected behaviour as the % length changes.

It's just that it was the "old topostats" default so wanted to keep it for tests etc. Do you know if any of your (@ns-rse) modified tests or @SylviaWhittle smoke tests use the `pruning_params.max_length` parameter value of "-1"?

If not this should be an easy remove.

Reply:
Yeah lots of my tests use `pruning_params.max_length = -1` but this can be changed easily. Any ideas on good values for the cats & rep ints?

Reply:
Group says to remove the % length as it doesn't make sense for mixed samples / conformations and requires back calculating to nm anyway for unknown structures. So I'll do that.

@SylviaWhittle ~10nm should be perfect (double the tip-convolved DNA width)

Reply:
Removed the -1 dependancy in [d0751ad](https://github.com/AFM-SPM/TopoStats/pull/898/commits/d0751adab8139465ada0551e8515544c24a3878b)